app/models/trade/annual_report_upload.rb

```diff
@@ -49,8 +49,8 @@ def validation_errors
     @primary_validation_errors || @secondary_validation_errors
   end
 
-  def sandbox
-    return nil if is_submitted?
+  def sandbox(tmp=false)
```

Comment:
This idea came along considering the fact that we are generating the log using a sidekiq job, so submission and changelog generation are asynchronous, meaning that the submission will proceed and the sandbox destroyed while the job it's still working.
This is used just in the changelog generation job and it's there to ensure that we can access the correct shipments class even if, in the meantime, the aru is submitted. 
After that, still just in the job, the sandbox it's called again with `tmp=true` along with `destroy` to ensure that everything has been destroyed correctly.

The only other way that came in mind to achieve this, is to use the standard `Trade::SandboxTemplate` class and use the ids of the shipments as reference, but maybe in this way it's more consistent.